### DISCUSSION TITLE
How to create Views with shared elements?

### DISCUSSION THREAD

**Discussion by BytePetter** [AvaloniaUI/Avalonia#12271] [Q&A]
Leaning on the idea of inheritance of C#-Classes, is it possible to create views that share common elements?

Lets say I want to have a view A.
Now every other view B that inherits from A should share some common visual elements. In my case, I would like to display an icon in the top right corner of the view. This should be defined in A, and displayed automatically in B. Is something like that possible? If yes, how?

**Comment by wieslawsoltes** [comment]
You can do templated control for base view A and use it as host for other controls.

**Comment by mysteryx93** [comment]
Sounds like you want a ContentControl that contains another control.

Similar to what I did with [MediaPlayerUI](https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Avalonia/Styles/Default/Theme.axaml), host control has the toolbar and all that, and you put a media player control inside it for the actual video playback of your choice.

[Theme file is here](https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Avalonia/Styles/Default/Theme.axaml)
[MediaPlayer container is here](https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Avalonia/MediaPlayer.cs)

With this approach, you need to insert both controls on every use. If you want to define your control as being both components already working together, you can use @wieslawsoltes  approach, for shorter syntax on usage.
